src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -45,6 +45,8 @@ def __init__(
         group_manager: GroupManager,
         migration_status_refresher: 'MigrationStatusRefresher',
         principal_grants: PrincipalACL,
+        hiveserde_in_place_migrate: HiveSerdeType | None = None,
+        mounts_crawler: Mounts | None = None,
```

Comment:
```suggestion
        mounts_crawler: Mounts,
```
No optional components are allowed via constructor injection. If you need to do something specific with mounts, then create another property in GlobalContext and depend on mounts and table migrator. Otherwise - always inject mounts without a chance of them being optional

Reply:
Moved the mounts out of the constructor and make it an optional parameter of `migrate_tables` function. The mounts will only be used if in place migrate the hiveserde tables